notebooks/02_model/rbm_deep_dive.ipynb

```diff
@@ -0,0 +1,1449 @@
+    "\n",
+    "\n",
+    "|  $X$   |$i_1$  |$i_2$  |$i_3$  |  ... |$i_m$  | \n",
+    "|-----|-------|-------|-------|------|-------|\n",
+    "|$u_1$|5      |0      |2      |0 ... |1      |\n",
+    "|$u_2$|0      |0      |3      |4 ... |0      |\n",
+    "|...  |...    |...    |...    |...   |...    |\n",
+    "|$u_m$|3      |3      |0      |5...  |2      |\n",
+    "\n",
+    "\n",
+    "where zeroes denote unrated items. In a nutshell, the recommender task is to \"fill in\" the missing ratings (later we will see that in practice this is not the only criteria to recommend a product). The classical approach to this problem is called matrix factorization: the basic idea is to decompose $X$ into a user ($P$) and item ($Q$) matrix, such that $X = Q^T P$. The dimensions of the two matrices are $dim(Q) = (f, n)$ and $dim(P)= (f,m)$ where $f \\le m,n$ is the number of latent factors, e.g. the genre of a movie, the type of food etc... and it is an hyperparameter of the model, for more details see the [ALS notebook](notebooks/02_modeling/als_deep_dive.ipynb). By learning $Q$ and $P$ we try to reproduce a particular instance of $X$ (provided by the available data) and use this information to fill up the missing matrix elements. \n",
+    "\n",
+    "The RBM approach is to look at $X$ as a particular realization (sample) of a more general process; instead of learning a specific $X$, we try to learn the matrix distrbution from which $X$ has been sampled form. Efectively, we learn the typical distirbution of *tastes* (i.e. latent factors) and use this information to *generate* new ratings. For this reaoson, this class of neural network models is also called **generative**. Consider the following example: imagine you are given the  income distribution per age window of a particular country (this is easy to find from goverments data), then we could fix the age window and *generate* virtual citizens with various incomes by sampling from this distibution.   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1.2 Model \n",
+    "\n",
+    "The central quantity of every SM model is the [Boltzmann distribution ](https://en.wikipedia.org/wiki/Boltzmann_distribution); this can be seen as the least biased probability distribution on a given probability space $\\Sigma$ and can be obtained using a maximum entropy principle on the space of distributions over $\\Sigma$. Its typical form is: \n",
+    "\n",
+    "$P = \\frac{1}{Z} \\, e^{- \\beta \\, H}$, \n",
```

Comment:
Two dollar signs can be make the equation aligned to the center, which may display better.